src/NHapi.Base/Parser/ParserBase.cs

```diff
@@ -1,415 +1,410 @@
+        /// </summary>
+        /// <throws>  EncodingNotSupportedException if the requested encoding is not.  </throws>
+        /// <summary>      supported by this parser.
+        /// </summary>
+        protected internal abstract string DoEncode(IMessage source, string encoding);
+
+        /// <summary> Called by encode(Message) to perform implementation-specific encoding work.
+        ///
+        /// </summary>
+        /// <param name="source">a Message object from which to construct an encoded message string.
+        /// </param>
+        /// <returns> the encoded message.
+        /// </returns>
+        /// <throws>  HL7Exception if the data fields in the message do not permit encoding.  </throws>
+        /// <summary>      (e.g. required fields are null).
+        /// </summary>
+        /// <throws>  EncodingNotSupportedException if the requested encoding is not.  </throws>
+        /// <summary>      supported by this parser.
+        /// </summary>
+        protected internal abstract string DoEncode(IMessage source);
+
+        /// <summary> Called by parse() to perform implementation-specific parsing work.
+        ///
+        /// </summary>
+        /// <param name="message">a String that contains an HL7 message.
+        /// </param>
+        /// <param name="version">the name of the HL7 version to which the message belongs (eg "2.5").
+        /// </param>
+        /// <returns> a HAPI Message object parsed from the given String.
+        /// </returns>
+        /// <throws>  HL7Exception if the message is not correctly formatted.   </throws>
+        /// <throws>  EncodingNotSupportedException if the message encoded.  </throws>
+        /// <summary>      is not supported by this parser.
+        /// </summary>
+        protected internal abstract IMessage DoParse(string message, string version);
+
+        /// <summary>
+        /// Note that the validation context of the resulting message is set to this parser's validation
+        /// context. The validation context is used within Primitive.setValue().
+        /// </summary>
+        /// <param name="theName">name of the desired structure in the form XXX_YYY.</param>
+        /// <param name="theVersion">HL7 version (e.g. "2.3").</param>
+        /// <param name="isExplicit">
+        /// true if the structure was specified explicitly in MSH-9-3, false if it
+        /// was inferred from MSH-9-1 and MSH-9-2.  If false, a lookup may be performed to find
+        /// an alternate structure corresponding to that message type and event.
+        /// </param>
+        /// <returns>a Message instance.</returns>
+        /// <exception cref="HL7Exception">Thrown when the version is not recognized or no appropriate class can be found or the Message.</exception>
+        protected internal virtual IMessage InstantiateMessage(string theName, string theVersion, bool isExplicit)
+        {
+            IMessage result = null;

```

Comment:
not sure why thi is ever set to null, looking at the code below it cant be null.

Reply:
done